docs/decisions/0006-api-specification-for-enterprise-mfes.rst

```diff
@@ -0,0 +1,308 @@
+
+This API endpoint works by iterating over all subsidy access policies associated with
+the enterprise customer for the specified learner to understand which subsidy access policies are indeed
+redeemable. It does this by calling the ``can_redeem`` API endpoint in ``enterprise-subsidy`` for each active
+subsidy access policy. In the event multiple subsidy access policies are found, this API endpoint chooses
+the preferred subsidy based on defined business rules.
+
+This API endpoint differs from the one spec'd in the `0003 Initial API Specification`_ in that
+it makes a decision of which single subsidy access policy should be redeemed for a given course in the event
+a learner has multiple redeemable subsidy access policies. The API endpoint that's already spec'd returns a
+list of redeemable policies, which would mean the client  (e.g., enterprise learner portal) still requires business
+logic to make a choice of which redeemable policy to attempt a redemption. This new API endpoint would thus remove the
+need for such business logic in the client given the subsidy access policy choice is abstracted into the API layer instead.
+
+The redemption fulfillment status is retrieved from the ``/api/v1/transactions/<transaction_uuid>/`` API endpoint in ``enterprise-subsidy``,
+which returns an individual transaction and its current state (i.e., ``created``, ``pending``, ``committed``).
+
+*Inputs (query parameters):*
+
+* ``lms_user_id``
+
+*Outputs:*
+
+For each course run associated with the specified ``content_key``:
+
+* A single, redeemable subsidy access policy (if any).
+* Redemption status of the single, redeemable subsidy access policy (if any).
+* List of error(s) for why there is no redeemable subsidy access policy.
+
+Sample API responses
+^^^^^^^^^^^^^^^^^^^^
+
+::
+
+  [
+    {
+      "course_run_key": "course-v1:ImperialX+dacc003+3T2019",
+      "redemption": {
+        "uuid": "26cdce7f-b13d-46fe-a395-06d8a50932e9",
+        "status": "fulfilled",
+        "policy_redemption_status_url": "/api/v1/policy/56744a36-93ac-4e6c-b998-a2a1899f2ae4/redemptions/26cdce7f-b13d-46fe-a395-06d8a50932e9/",
+        "courseware_url": "https://courses.edx.org/courses/course-v1:ImperialX+dacc003+3T2019/courseware/",
+        "errors": []
+      },
+      "subsidy_access_policy": {
+        "uuid": "56744a36-93ac-4e6c-b998-a2a1899f2ae4",
+        "policy_redemption_url": "/api/v1/policy/56744a36-93ac-4e6c-b998-a2a1899f2ae4/redeem/",
+        "policy_type": "LearnerCreditAccessPolicy",
+        "description": "Learner credit access policy",
+        "active": true,
+        "catalog_uuid": "14f701ea-7e0b-4a4e-bbda-f295e40c7bf1",
+        "subsidy_uuid": "7801b0ef-b1c2-4f3a-97fa-121f0bce48be",
+        "access_method": "direct",
+        "spent_limit": 10000,
+        "per_learner_spend_limit": 200,
+        "remaining_balance": 9500,
+        "remaining_balance_for_learner": 200
+      },
+      "errors": []
```

Comment:
Maybe this is really `reasons`?  `errors` we'll want to reserve for cases where e.g. fulfillment is in a pending or failed state.